torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -913,7 +913,10 @@ void CudaCodeGen::call(const std::vector<CallArg>& args) {
     ptr_to_args[buffer_args.size()] = &rand_seed;
     ptr_to_args[buffer_args.size() + 1] = &rand_offset;
   }
-
+  const auto prior_device = at::cuda::current_device();
```

Comment:
Good catch!

Since you are now using this->device(), it might be a good idea to check that device() can only be kCuda somewhere in "initialize()".